include/drivers/gpio.h

```diff
@@ -32,6 +32,320 @@ extern "C" {
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_LOW_0              (1 << 15)
+
+/** Trigger detection on input state is (or transitions to) physical high or
+ * logical 1 level.
+ *
+ * This is a component flag that should be combined with other
+ * `GPIO_INT_*` flags to produce a meaningful configuration.
+ */
+#define GPIO_INT_HIGH_1             (1 << 16)
+
+/** Trigger GPIO interrupt on pin rising edge. */
+#define GPIO_INT_EDGE_RISING        (GPIO_INT_ENABLE \
+				     | GPIO_INT_EDGE \
+				     | GPIO_INT_HIGH_1)
+
+/** Trigger GPIO interrupt on pin falling edge. */
+#define GPIO_INT_EDGE_FALLING       (GPIO_INT_ENABLE \
+				     | GPIO_INT_EDGE \
+				     | GPIO_INT_LOW_0)
+
+/** Trigger GPIO interrupt on pin rising or falling edge. */
+#define GPIO_INT_EDGE_BOTH          (GPIO_INT_ENABLE \
+				     | GPIO_INT_EDGE \
+				     | GPIO_INT_LOW_0 \
+				     | GPIO_INT_HIGH_1)
+
+/** Trigger GPIO interrupt on pin physical level low. */
+#define GPIO_INT_LEVEL_LOW          (GPIO_INT_ENABLE \
+				     | GPIO_INT_LEVEL \
+				     | GPIO_INT_LOW_0)
+
+/** Trigger GPIO interrupt on pin physical level high. */
+#define GPIO_INT_LEVEL_HIGH         (GPIO_INT_ENABLE \
+				     | GPIO_INT_LEVEL \
+				     | GPIO_INT_HIGH_1)
+
+/** Trigger GPIO interrupt on pin state change to logical level 1. */
+#define GPIO_INT_EDGE_TO_ACTIVE     (GPIO_INT_ENABLE \
+				     | GPIO_INT_LEVELS_LOGICAL \
+				     | GPIO_INT_EDGE \
+				     | GPIO_INT_HIGH_1)
+
+/** Trigger GPIO interrupt on pin state change to logical level 0. */
+#define GPIO_INT_EDGE_TO_INACTIVE   (GPIO_INT_ENABLE \
+				     | GPIO_INT_LEVELS_LOGICAL \
+				     | GPIO_INT_EDGE \
+				     | GPIO_INT_LOW_0)
+
+/** Trigger GPIO interrupt on pin logical level 0. */
+#define GPIO_INT_LEVEL_INACTIVE     (GPIO_INT_ENABLE \
+				     | GPIO_INT_LEVELS_LOGICAL \
+				     | GPIO_INT_LEVEL \
+				     | GPIO_INT_LOW_0)
+
+/** Trigger GPIO interrupt on pin logical level 1. */
+#define GPIO_INT_LEVEL_ACTIVE       (GPIO_INT_ENABLE \
+				     | GPIO_INT_LEVELS_LOGICAL \
+				     | GPIO_INT_LEVEL \
+				     | GPIO_INT_HIGH_1)
+
+/** @} */
+
+/** Enable GPIO pin debounce. */
+#define GPIO_INT_DEBOUNCE           (1 << 17)
+
+/**
+ * @name GPIO drive strength flags
+ * The `GPIO_DS_*` flags are used with `gpio_pin_configure` or
+ * `gpio_port_configure`, to specify the drive strength configuration of a GPIO
+ * pin.
+ *
+ * The drive strength of individual pins can be configured
+ * independently for when the pin output is low and high.
+ *
+ * The `GPIO_DS_*_LOW` enumerations define the drive strength of a pin
+ * when output is low.
+
+ * The `GPIO_DS_*_HIGH` enumerations define the drive strength of a pin
+ * when output is high.
+ *
+ * The interface supports two different drive strengths:
+ * `DFLT` - The lowest drive strength supported by the HW
+ * `ALT` - The highest drive strength supported by the HW
```

Comment:
Nordic and SX1509B differ in whether the peripheral-defined default is lower or higher strength than the alternative setting.   On Nordic it's standard and high; on SX1509B it's normal and reduced.  I suppose it's prior art (introduced for Nordic) so you have to do something, but IMO this is the sort of thing that doesn't belong in a generic driver API.  There's really no way this can be used portably.

Reply:
As I mentioned in the PR description
> Any functional changes not mentioned here will be addressed in subsequent PRs. Specifically this PR doesn't modify flags related to pin drive strength, debouncing neither attempts to introduce functions operating on ports.

Linux DTS nodes define drive strength by specifying sink or source current in mA. To have compliant DTS description we would need to do the same, but maybe it's not worth it. I proposed to handle this issue later. It has no influence on the modifications introduced by this PR.